server/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -190,9 +194,18 @@ private static void parseSearchSource(final SearchSourceBuilder searchSourceBuil
         if (request.hasParam("from")) {
             searchSourceBuilder.from(request.paramAsInt("from", 0));
         }
-        int size = request.paramAsInt("size", -1);
-        if (size != -1) {
-            setSize.accept(size);
+        if (request.hasParam("size")) {
+            int size = request.paramAsInt("size", -1);
+            if (size != -1) {
+                setSize.accept(size);
+            } else {
+                deprecationLogger.deprecate(
+                    DeprecationCategory.API,
+                    "search-api-size-1",
+                    "Using search size of -1 is deprecated and will be removed in future versions. Instead, don't use the `size` parameter "
+                        + "if you don't want to set it explicitely."
```

Comment:
I wonder if a deprecation warning is needed here, it seems like a bug that somebody provides a negative value for it? Also aren't you logging whenever size was not specified too?

Reply:
You are right the we have rejected `-1` in the setter in SearchSourceBuilder here for a long time, however currently when `-1` is specified in the rest request parameters, the value is dropped and nothing breaks. Not deprecating at least for a few minors in 7.x might look like a breaking change to users that for whatever reason use `-1` at the moment to indicate something like "not-set". I agree we should remove this in 8.0, I would do that in a follow up on master when the deprecation here is backported.


> Also aren't you logging whenever size was not specified too?

That shouldn't happen because of `if (request.hasParam("size"))` being false then, or am I missing something?

Reply:
I had missed the conditional!